### PR TITLE
feat: prompt to save generator on exit

### PR DESCRIPTION
**Pull request by cristianoventura**
## Description

This PR triggers the dialog to save the generator (if unsaved changes are present) when closing the main window.

## How to Test

- Open the generator
- Make modifications to it
- Close the main window
- Check that the save dialog is triggered

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [x] I have run linter locally (`npm run lint`) and all checks pass.
- [x] I have run tests locally (`npm test`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/80506356-bacf-4b93-914f-09c6c4435ac2

## Related PR(s)/Issue(s)

Resolves https://github.com/grafana/k6-studio/issues/239

### CODE REVIEW

File: src/components/Layout/Layout.tsx [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+  const location = useLocation()
 
   const handleVisibleChange = (index: number, visible: boolean) => {
     if (index !== 1) return
     setIsSidebarExpanded(visible)
   }
 
+  useEffect(() => {
+    window.studio.app.changeRoute(location.pathname)
+  }, [location])
```
Comment (cristianoventura) [inline src/components/Layout/Layout.tsx:22]: I'm keeping track of the routes on the electron side so we know which route the `close` event in the main window came from.

**Review by Llandy3d — DISMISSED** [review]
Reviewing from the functionality standpoint, works great 🙌

**Review by going-confetti — DISMISSED** [review]
LGTM. Thought the copy could be improved for this specific case. Maybe something like this will work in both cases:
> You have unsaved changes in the generator which will be lost if you proceed

Although I see that it's not ideal either

**Comment by Llandy3d** [comment]
what about the same text plus `upon leaving` 👀 

It should work both for leaving the page and leaving the application!

**Comment by cristianoventura** [comment]
@Llandy3d I liked your suggestion. Just pushed a change with it!

**Review by Llandy3d — APPROVED** [review]
🙌

**Pull request merged**